### ISSUE TITLE
Improve orientationC3<FT> result when 2 of the 4 points are completely coincide

### ISSUE THREAD

**Issue by AlexLiogky** [CGAL/cgal#8618]
## Issue Details
Issue type: suggestion how to improve the wrong result in orientationC3<FT>

See file `cgal/Cartesian_kernel/include/CGAL/predicates/kernel_ftC3.h` .

If numeric type FT meets the IEEE 754 standard and 2 points from the set (p, q, r, s) are exactly the same, then it is possible to ensure that [orientationC3<FT>](cgal/Cartesian_kernel/include/CGAL/predicates/kernel_ftC3.h`) always returns the correct result CGAL::ZERO. To do this, it is enough to replace the order of values in the call inside orientationC3<FT>
```C++
return sign_of_determinant<FT>(qx-px,rx-px,sx-px,
                               qy-py,ry-py,sy-py,
                               qz-pz,rz-pz,sz-pz);
``` 
with a transposed one
```C++
return sign_of_determinant<FT>(qx-px,qy-py,qz-pz,
                               rx-px,ry-py,rz-pz,
                               sx-px,sy-py,sz-pz); 
``` 
In this case, the order of the variables will be consistent with the [determinant](cgal/Kernel_23/include/CGAL/determinant.h) function 
```C++
RT
determinant(
 const RT& a00,  const RT& a01,  const RT& a02,
 const RT& a10,  const RT& a11,  const RT& a12,
 const RT& a20,  const RT& a21,  const RT& a22)
{
// First compute the det2x2
  const RT m01 = a00*a11 - a10*a01;
  const RT m02 = a00*a21 - a20*a01;
  const RT m12 = a10*a21 - a20*a11;
// Now compute the minors of rank 3
  const RT m012 = m01*a22 - m02*a12 + m12*a02;
  return m012;
}
```
In this case, if there are two identical points among (p, q, r, s) inside the determinant function, either one of the rows will be zero, leading to m012 = 0.0, or there will be 2 identical rows, which, due to the order of calculations, will lead to one of the numbers (m01, m02, m12) will turn out to be equal to 0.0, and absolute value of the other two numbers from (m01, m02, m12) will be exactly equal calculating m012 will give an exact 0.0 in machine arithmetic.

If there are no other reasons why such a transposition inside orientationC3<FT> should not be done, then it makes sense to do it.

## Source Code

```C++
#include <CGAL/Simple_cartesian.h>
#include <CGAL/Triangulation_3.h>
#include <CGAL/assertions.h>

using namespace CGAL;
using Point_3 = typename CGAL::Simple_cartesian<double>::Point_3;

int main(){
	static_assert(sizeof(double) == 8);
	double d[12] ={
		0x1.ceb594p+5, //57.838661193847656
		0x1.8c9f8cp+5, //49.577903747558594
		0x1.6457a4p+5, //44.542793273925781
		0x1.2c5b8ap+5, //37.544696807861328
		0x1.98d706p+5, //51.104991912841797
		0x1.61c26ap+5, //44.219928741455078
		0x1.61eac8p+5, //44.239639282226562
		0x1.141ddcp+5, //34.514579772949219
		0x1.b35be6p+5, //54.419872283935547
		0x1.2c5b8ap+5, //37.544696807861328
		0x1.98d706p+5, //51.104991912841797
		0x1.61c26ap+5, //44.219928741455078
	};
	Point_3 p(d[0], d[1], d[2]), q(d[3], d[4], d[5]), r(d[6], d[7], d[8]), s(d[9], d[10], d[11]);
	// Here q == s
	Orientation o = orientation(p, q, r, s); //here we get CGAL::NEGATIVE
	CGAL_precondition(o == CGAL::ZERO);
	
	return 0;
}
```

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Ubuntu 23.10 64 bits
* Compiler: GCC 13.2.0 x86_64-linux-gnu
* Release or debug mode: Debug
* Specific flags used (if any): -
* CGAL version: 5.6.2
* Boost version: 1.74.0
* Other libraries versions if used (Eigen, TBB, etc.): -

**Comment by mglisse** [comment]
That's an interesting idea, but I am not convinced it is that helpful: you cannot rely on the result of `orientation` for `Simple_cartesian<double>` anyway.
About the transposition, I'd be more interested in whether `orientation(a,b,c,d)` and `orientation(a,b,c,e)` have a lot of computations in common (that a compiler might be able to deduplicate if the 2 calls are consecutive, which might happen in the structural filter of the location in Delaunay triangulations). I didn't check which orientation of the matrix gives that.

**Comment by AlexLiogky** [comment]
Yes, you are absolutely right, you cannot rely on `orientation` for `Simple_cartesian<double>`. 
It seems that it is the current implementation that potentially allows the compiler to reuse common calculations in the scenario you described. That's why I'm closing the issue, thank you.

Meanwhile, I actually got this idea when I tried to remesh `Surface_mesh<Simple_cartesian<double>::Point_3>` while preserving the position of some vertices. To do this, I tried to use `PMP::surface_Delaunay_remeshing`, but found that this function does not support `vertex_is_constrained_map` and then just copied its contents and added the `domain.add_corners` call. Then the call to `CGAL::make_mesh_3` fell on checking that `insert_corner() -> 620: dt.is_vertex()` because it did not recognize the vertex. However, even though the modification I proposed for the `orientation` function removes failure in this particular place, then failures still occur in the coplanarity tests, so this modification really makes little sense
